### PR TITLE
Add sol-netctl API to get extra service properties V3

### PR DESCRIPTION
**Pull request by guchaojie**
This PR is for to improve getting extra service properties for user
All the properties provided by Connman D-Bus Hierarchy

Signed-off-by: guchaojie <chao.jie.gu@intel.com>

### CODE REVIEW

**Comment by guchaojie** [comment]
@cabelitos @barbieri  Sorry for delaying update PR due to Chinese Mid-autumn Festival. Please have a review , thanks!

File: src/lib/comms/sol-netctl-impl-connman.c [hunk 1]
```diff
@@ -182,6 +252,95 @@ sol_netctl_service_get_state(const struct sol_netctl_service *service)
...
+{
+    SOL_NULL_CHECK(service, NULL);
+
+    return &service->security;
+}
+
+SOL_API bool
+sol_netctl_service_is_favorite(const struct sol_netctl_service *service)
+{
+    SOL_NULL_CHECK(service, NULL);
```
Comment (ceolin) [inline src/lib/comms/sol-netctl-impl-connman.c]: you are returning a wrong type, should return `FALSE`

File: src/lib/comms/sol-netctl-impl-connman.c [hunk 2]
```diff
@@ -182,6 +252,95 @@ sol_netctl_service_get_state(const struct sol_netctl_service *service)
...
+
+    return service->favorite;
+}
+
+SOL_API bool
+sol_netctl_service_is_immutable(const struct sol_netctl_service *service)
+{
+    SOL_NULL_CHECK(service, NULL);
```
Comment (ceolin) [inline src/lib/comms/sol-netctl-impl-connman.c]: ditto

File: src/lib/comms/sol-netctl-impl-connman.c [hunk 3]
```diff
@@ -328,8 +487,317 @@ get_service_ip(sd_bus_message *m,
     } while (1);
 
 end:
-    if (r == 0)
+    exit = sd_bus_message_exit_container(m);
```
Comment (ceolin) [inline src/lib/comms/sol-netctl-impl-connman.c]: you don't need another variable here `exit`. You can re-use `r`, the return result using exit or  r here will be the same.

File: src/lib/comms/sol-netctl-impl-connman.c [hunk 4]
```diff
@@ -328,8 +487,317 @@ get_service_ip(sd_bus_message *m,
...
+            SOL_INT_CHECK(r, < 0, r);
+        }
+
         r = sd_bus_message_exit_container(m);
+        SOL_INT_CHECK(r, < 0, r);
+    } while (1);
+
+end:
+    exit = sd_bus_message_exit_container(m);
+    SOL_INT_CHECK(exit, < 0, exit);
```
Comment (ceolin) [inline src/lib/comms/sol-netctl-impl-connman.c]: same commet about exit here.

**Pull request closed without merging**